homeassistant/helpers/recorder.py

```diff
@@ -56,3 +64,42 @@ async def async_wait_recorder(hass: HomeAssistant) -> bool:
     if DOMAIN not in hass.data:
         return False
     return await hass.data[DOMAIN].db_connected
+
+
+@contextmanager
+def session_scope(
+    *,
+    hass: HomeAssistant | None = None,
+    session: Session | None = None,
+    exception_filter: Callable[[Exception], bool] | None = None,
+    read_only: bool = False,
+) -> Generator[Session]:
+    """Provide a transactional scope around a series of operations.
+
+    read_only is used to indicate that the session is only used for reading
+    data and that no commit is required. It does not prevent the session
+    from writing and is not a security measure.
+    """
+    # pylint: disable-next=import-outside-toplevel
+    from homeassistant.components.recorder.util import get_instance
```

Comment:
Can we avoid the late import, or at least only do it conditionally so it doesn't have to be checked as much?

Reply:
We could move the `get_instance` function to the recorder helper to avoid it